web/base_viewer.js

```diff
@@ -127,6 +127,20 @@ function isSameScale(oldScale, newScale) {
   return false;
 }
 
+function onModifiedAnnotationStorage() {
+  window.addEventListener("beforeunload", beforeUnload);
```

Comment:
This function is not called anywhere else and trivial, so wouldn't it be easier to just inline it below like this (same for the reset function)?

```javascript
annotationStorage.onModified = function() {
  window.addEventListener("beforeunload", beforeUnload);
};
```

That way the relevant code is more in one place which makes it easier to follow in my opinion.